### ISSUE TITLE
Support for provisioned throughput SKU 

### ISSUE THREAD

**Issue by zioproto** [Azure/terraform-azurerm-openai#70]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Feature description https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/provisioned-throughput

Pending on azurerm provider support: 
https://github.com/hashicorp/terraform-provider-azurerm/issues/23536

### New or Affected Resource(s)/Data Source(s)

azurerm_cognitive_deployment

**Comment by zioproto** [comment]
This should work already like this:

```
[..CUT..]
  deployment = {
    "chat_model" = {
      name          = "gpt-4"
      model_format  = "OpenAI"
      model_name    = "gpt-4"
      model_version = "0613"
      scale_type    = "ProvisionedManaged"
      capacity      = 100
    },
[..CUT..]
```

**Comment by shancodec** [comment]
Yes this works. However not sure why Terraform is still mentioning the values as "P3"

**Comment by zioproto** [comment]
> Yes this works. However not sure why Terraform is still mentioning the values as "P3"

Hello, could you please provide more context? I am not able to understand.

Could you provide a screenshot or cut and paste the relevant part of the Terraform plan ?

thanks
